src/main/java/qna/domain/user/User.java

```diff
@@ -1,19 +1,38 @@
-package qna.domain;
-
-import qna.UnAuthorizedException;
+package qna.domain.user;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import qna.domain.EntityHistory;
+import qna.exception.UnAuthorizedException;
 
 import java.util.Objects;
 
-public class User {
-    public static final GuestUser GUEST_USER = new GuestUser();
+@Entity
+@Table(name = "user", uniqueConstraints = @UniqueConstraint(columnNames = {"userId"}))
+public class User extends EntityHistory {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    private String userId;
-    private String password;
-    private String name;
+
+    @Column(length = 50)
     private String email;
 
-    private User() {
+    @Column(nullable = false, length = 20)
+    private String name;
+
+    @Column(nullable = false, length = 20)
+    private String password;
+
+    @Column(nullable = false, length = 20)
```

Comment:
`uniqueConstraints` 옵션과 아래 코드는 어떤 차이가 있을까요?

```suggestion
    @Column(unique = true, nullable = false, length = 20)
```

Reply:
실행되는 sql이 달라질 줄 알았는데 동일하네요. 

개별 컬럼에 대해서만 설정할 수 있는 대신 가독성이 개선되는게 전부려나요